packages/svelte/src/internal/client/reactivity/batch.js

```diff
@@ -717,6 +751,28 @@ function mark_effects(value, sources, marked, checked) {
 	}
 }
 
+/**
+ * When committing a fork, we need to trigger inspect effects so that
+ * any `$state.eager(...)` expressions update immediately. This
+ * function allows us to discover them
+ * @param {Value} value
+ * @param {Set<Effect>} effects
+ */
+function mark_eager_effects(value, effects) {
+	if (value.reactions !== null) {
+		for (const reaction of value.reactions) {
+			const flags = reaction.f;
+
+			if ((flags & DERIVED) !== 0) {
+				mark_eager_effects(/** @type {Derived} */ (reaction), effects);
+			} else if ((flags & EAGER_EFFECT) !== 0) {
+				set_signal_status(reaction, DIRTY);
+				effects.add(/** @type {Effect} */ (reaction));
+			}
+		}
+	}
```

Comment:
This might read better as `if (value.reactions === null) return;`, just reduces the nesting